dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -1476,6 +1476,24 @@ protected boolean useExperimentalAnalyticsQueryEngine() {
         || this.settingsService.getCurrentSettings().getUseExperimentalAnalyticsQueryEngine();
   }
 
+  /**
+   * Transforms the query item filters into an "and" separated SQL string. For instance, if the
+   * query item has filters with values "a" and "b" and the operator is "eq", the resulting SQL
+   * string will be "column = 'a' and column = 'b'". If the query item has no filters, an empty
+   * string is returned.
+   *
+   * @param item the {@link QueryItem}.
+   * @param columnName the column name.
+   * @return the SQL string.
+   */
+  protected String extractFiltersAsSql(QueryItem item, String columnName) {
+    return item.getFilters().stream()
+        .map(
+            f ->
+                "%s %s %s".formatted(columnName, f.getOperator().getValue(), getSqlFilter(f, item)))
```

Comment:
I'm unsure if this will work with `IN` query filter and `NV` value. Those two need special handling.